templates/pkg/crd_sdk.go.tpl

```diff
@@ -188,22 +189,38 @@ func (rm *resourceManager) newCreateRequestPayload(
 func (rm *resourceManager) sdkUpdate(
 	ctx context.Context,
 	r *resource,
+	diffReporter *ackreporter.Reporter,
 ) (*resource, error) {
 {{- if .CRD.Ops.Update }}
 	input, err := rm.newUpdateRequestPayload(r)
 	if err != nil {
 		return nil, err
 	}
-{{ $setCode := GoCodeSetUpdateOutput .CRD "resp" "ko.Status" 1 }}
+
+    {{ if .CRD.HasCustomUpdateOperations }}
+
+    {{ $setCode := GoCodeSetUpdateOutput .CRD "" "ko.Status" 1 }}
+    {{ GoCodeCustomUpdate .CRD "ctx" "r" "diffReporter" 1 }}
+    // Merge in the information we read from the API call above to the copy of
+    // the original Kubernetes object we passed to the function
+    ko := r.ko.DeepCopy()
+    {{ $setCode }}
+
+    {{ else }}
```

Comment:
Here, I was thinking you could just use `{{ end }}` instead of  `{{ else }}`. That way, you could place a good chunk of the static Go code generation here, for example, you could do something like this:

```
    {{ if .CRD.HasCustomUpdateOperations }}
		for _, diff := range diffReporter.Differences {
			switch diff.Path {
        {{- range $diffPath, $customMethod := .CRD.GetCustomUpdateOperations }}
			case "{{ $diffPath }}":
				if err := rm.UpdateReplicaCount(ctx, r, diffReporter); err != nil {
					return nil, err
				}
         {{- end }}
			}
    {{ end }}
```

And you wouldn't need any of the `CRD.GoCodeCustomOutput` method at all...

Reply:
Thanks for the suggestion. I have applied it in the code.